### PR TITLE
Add a test for sortBy

### PR DESCRIPTION
**Pull request by johnynek**
This adds a test for the possible bug in #1593 but does not find any issue.

### CODE REVIEW

**Comment by piyushnarang** [comment]
👍

File: scalding-core/src/test/scala/com/twitter/scalding/TypedPipeTest.scala [hunk 1]
```diff
@@ -121,6 +121,38 @@ class TypedSumByKeyTest extends WordSpec with Matchers {
   }
 }
 
+class TypedPipeSortByJob(args: Args) extends Job(args) {
+  TypedPipe.from(TypedText.tsv[(Int, Float, String)]("input"))
+    .groupBy(_._1)
+    .sortBy(_._2)
+    .mapValues(_._3)
+    .sum
```
Comment (ulyssepence) [inline scalding-core/src/test/scala/com/twitter/scalding/TypedPipeTest.scala:129]: `.sum` only requires associativity?

Comment (ulyssepence) [inline scalding-core/src/test/scala/com/twitter/scalding/TypedPipeTest.scala:129]: Oh right, this uses a `Semigroup`. NVM.

**Comment by ulyssepence** [comment]
+1

**Pull request merged**